src/api/java/com/minecolonies/api/blocks/AbstractBlockBarrel.java

```diff
@@ -24,6 +24,10 @@ public AbstractBlockBarrel(final Properties properties)
 
     public static BlockState changeStateOverFullness(@NotNull final AbstractTileEntityBarrel te, @NotNull final BlockState blockState)
     {
+        if (blockState.getBlock() != ModBlocks.blockBarrel)
+        {
+            return blockState;
+        }
```

Comment:
If this logic should only work for the Barrel block, then this needs to be moved to its override, instead of this if-switch being here.

Reply:
done